File: test/end-to-end/Graphs/Ruby/data/unweighted_node.rb

##
# An unweighted, undirected node in a graph.
##
class UnweightedNode
    attr_accessor :data
    attr_accessor :neighborNodes

    def initialize(data)
        self.data = data
        self.neighborNodes = []
    end

    def add_neighbor(node)
        self.neighborNodes.push(node)
    end

    def get_neighbors_in_order()
        return self.neighborNodes
    end
end

File: test/end-to-end/Graphs/PHP/data/unweighted_node.php
<?php

/**
 * An unweighted, undirected node in a graph.
 */
class UnweightedNode {
    public $data;
    private $neighborNodes;

    function __construct($data) {
        $this->$data = $data;
        $this->$neighborNodes = [];
    }

    function addNeighbor($node) {
        array_push($this->$neighborNodes, $node);
    }

    function getNeighborsInOrder() {
        return $this->$neighborNodes;
    }
}
